src/main/kotlin/com/levi/featurespitter/service/EndPointService.kt

```diff
@@ -0,0 +1,82 @@
+package com.levi.featurespitter.service
+
+import org.springframework.stereotype.Service
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+import java.net.URI
+import com.levi.featurespitter.model.Store
+import com.levi.featurespitter.model.StoreSeason
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import org.springframework.beans.factory.annotation.Autowired
+
+@Service
+class EndPointService {
+
+
+	companion object {
+		fun getStores(): List<Store> {
+			val client = HttpClient.newBuilder().build();
+			val request = HttpRequest.newBuilder()
+				.uri(URI.create("http://134.209.29.209/v1/stores/?page=1"))
+				.header("apiKey", "nxzAAC1EQAAU7AGY8gUQnev_99A")
```

Comment:
This API key is hard-coded into the code which can pose a security concern, and adding to that, it is spread in multiple places in the code.

How would you improve this?

Reply:
To make it secure , We can store it in some properties file and read it from there in code. 
Again due to time constraint , I could not refactor the code, I wanted to place it at one place and refer at different places.

Reply:
👍

Reply:
Also, I just noticed that you are only fetching 1 page of stores...?!

Reply:
Yes Just for testing, I made it 1,But , I could have looped till the last record to pull all the records.

Reply:
Yes, but having to deal with all the pages would allow us to see how would you design your app to handle large amounts of data. You have design decisions that with 1 single page will not have much effect, but if you fetch all the pages will cause great distress for the app and the user.

Reply:
I understand that, we have different approaches to paginate. We can opt for query based pagination.